src/main/java/core/basesyntax/dao/impl/MessageDaoImpl.java

```diff
@@ -12,21 +14,62 @@ public MessageDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Message create(Message entity) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(entity);
+            transaction.commit();
+            return entity;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Couldn't create message: " + entity, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public Message get(Long id) {
-        return null;
+        try (Session session = factory.openSession()) {
+            return session.get(Message.class, id);
+        } catch (Exception e) {
+            throw new RuntimeException("Couldn't get message from DB where id: " + id, e);
+        }
     }
 
     @Override
     public List<Message> getAll() {
-        return null;
+        try (Session session = factory.openSession()) {
+            return session.createQuery("FROM Message", Message.class).getResultList();
```

Comment:
as far as I understand, we can call list() instead of getResultList(), because getResultList() will do the same thing under the "hood"
![image](https://user-images.githubusercontent.com/22300600/127899341-a793dc2f-221d-4bca-be4b-9c544731cc9b.png)